include/crpropa/module/Acceleration.h

```diff
@@ -0,0 +1,170 @@
+};
+
+
+/// @class SecondOrderFermi
+/// @brief  Implements scattering with centers moving in isotropic directions.
+///   All scatter centers have the same velocity.
+class SecondOrderFermi : public AbstractAccelerationModule
+{
+	double scatterVelocity;
+	std::vector<double> angle;
+	std::vector<double> angleCDF;
+
+	public:
+		SecondOrderFermi(double _scatterVelocity=.1 * crpropa::c_light, double stepLength = 1. * crpropa::parsec, unsigned int size_of_pitchangle_table=10000);
+		virtual crpropa::Vector3d scatterCenterVelocity(crpropa::Candidate *candidate) const;
+};
+
+
+/// @class DirectedFlowScattering
+/// @brief Scattering in a directed flow of scatter centers.
+/// @details Two of these region with different
+///		velocities can be used to create first order Fermi scenario.
+///		Thanks to Aritra Ghosh, Groningn University, for first work in 2017 on the
+///		shock acceleration in CRPropa leading to this module.
+class DirectedFlowScattering : public AbstractAccelerationModule
+{
+	private:
+		crpropa::Vector3d __scatterVelocity;
+	public:
+		DirectedFlowScattering(crpropa::Vector3d scatterCenterVelocity, double stepLength=1. * parsec);
+		virtual crpropa::Vector3d scatterCenterVelocity(crpropa::Candidate *candidate) const;
+};
+
+
+/// @class DirectedFlowOfScatterCenters
+/// @brief In a directed flow, the step length depend on the direction of the particles as headon collisions are more likely than tail=on collisions - propagating against the flow is harder.
+class DirectedFlowOfScatterCenters: public StepLengthModifier
+{
+	private:
+		Vector3d __scatterVelocity;
+	public:
+	DirectedFlowOfScatterCenters(const Vector3d &scatterCenterVelocity);
+	double modify(double steplength, Candidate* candidate);
+};
+
+
+/// @class QuasiLinearTheory
+/// @brief Scales the steplength according to quasi linear theory.
+/// @details Following quasi-linear theory [Schlickeiser1989], the mean free path \f$\lambda\f$ of a
+///  particle with energy \f$E\f$ and charge \f$Z\f$ in a field with turbulence spectrum
+///  \f$\frac{k}{k_{\min}}^{-q}\f$ is
+///   \f[ \lambda = {\left(\frac{B}{\delta B}\right)}^2 {\left(R_G\;  k_{\min}\right)}^{1-q} R_G \equiv \lambda_0 {\left( \frac{E}{1 EeV}\frac{1}{Z} \right)}^{2-q} \f]
+///  where \f$R_G = \frac{E}{B Z}\f$ is the gyro-radius of the
+///  particles.
+/// This class implements the rigidity dependent scaling factor used to modify
+/// the base step length.
+/// @par
+/// \b [Schlickeiser1989] R. Schlickeiser, Cosmic-Ray Transport and Acceleration. II.
+///  Cosmic Rays in Moving Cold Media with Application to Diffu-
+///  sive Shock Wave Acceleration, The Astrophysical Journal 336
+///  (1989) 264. doi:10.1086/167010.
+class QuasiLinearTheory : public StepLengthModifier
+{
+	private:
+		double __referenceEnergy;
+		double __turbulenceIndex;
+		double __minimumRigidity;
+
+	public:
+		QuasiLinearTheory(double referenecEnergy=1.*EeV, double turbulence_index=5./3, double minimumRigidity=0);
```

Comment:
For style convention, since we can use kwargs in python, shouldn't it be turbulenceIndex in camel case?